java/com/google/flatbuffers/Table.java

```diff
@@ -273,6 +284,320 @@ protected static int compareStrings(int offset_1, byte[] key, ByteBuffer bb) {
     }
     return len_1 - len_2;
   }
+
+  protected int __lookupByBooleanKey(int vectorFieldOffset, int keyFieldOffset, boolean queryKey,
+          boolean defaultValue ) {
+    byte key = (byte)(queryKey ? 1 : 0);
+    byte byteDefaultValue = (byte)(defaultValue ? 1 : 0);
+    return __lookupByByteKey(vectorFieldOffset, keyFieldOffset, key, byteDefaultValue);
+  }
+
+  protected int __lookupByByteKey(int vectorFieldOffset, int keyFieldOffset, int queryKey,
```

Comment:
Hi Wouter,
I'm agree with you using inheritance for technical concern is a bad design. But IMHO using generation to avoid it is not really better because for me generated code should be written as if I was writing it manually. Therefore what I propose is to extract those methods in a technical class and use delegation instead of inheritance.
What do you think about that?

Romain.